### DISCUSSION TITLE
AI Chat new feature

### DISCUSSION THREAD

**Discussion by leonardobetti** [elementary/triage#224] [Ideas]
HI all 😊 

It seems to me that AI Chat is going to be the next big trend in tech for a few years, and it's a trend that open source organizations should get involved in. I've created this basic bash script that installs [Shell Genie](https://github.com/dylanjcastillo/shell-genie) - an AI Chat tool that uses Open AI Chat GPT. 

Basically, the script check dependencies, install Shell Genie (or _**Elementary AI Chat**_), give the user a link to generate an API Key and also provide an example how to use the tool. This script makes it easy for users to generate an API key for the ChatGPT API and install Shell Genie using pip and pipx. It also creates an alias so that users can use the chat command instead of typing out shell-genie ask every time. I'm sharing because this script is just an example of how we could enable Chat GPT-3 within Elementary OS, so that users can use Chat GPT within the terminal.

I think it would be a great idea to include the option to enable Chat GPT during the onboarding screens, maybe ask for API Keys during the onboarding if the user wants to enable 🤔 However, I'm not sure about all the legal and regulatory / license requirements that need to be considered. Unfortunately, I'm not skilled enough to create a proper app just yet, since I still need to learn a bit of Vala and GTK4 :) But I hope that this script will be useful as an idea. I'd love to hear any feedback, more ideas etc.

```bash
#!/bin/bash

# Welcome message
echo "Welcome to Elementary AI Chat"

# Check if pip and pipx are installed
if ! command -v pip >/dev/null 2>&1; then
  echo "Installing dependencies..."
  sudo apt-get install python3-pip
fi

if ! command -v pipx >/dev/null 2>&1; then
  echo " ..."
  curl https://bootstrap.pypa.io/get-pipx.py | python3
fi

# Check if the user wants to enable the feature
echo "Would you like to enable the shell-genie feature? (y/n)"
read user_input

# If the user does not want to enable the feature, exit the script
if [[ $user_input != "y" ]]; then
  echo "Exiting..."
  exit 1
fi

# Install Shell Genie
echo "Elementary AI Chat.."
pipx install shell-genie

# Get the ChatGPT Key
echo "You need to generate an API Key to use Chat GPT-3. Please visit https://platform.openai.com/account/api-keys and generate your key"

# Create an alias for the shell-genie ask command
echo "Creating an alias for the shell-genie ask command..."
echo "alias chat='shell-genie ask'" >> ~/.bash_aliases

# Success message
echo "Elementary AI Chat has been successfully installed. Please select openai-gpt3. After that, provide your API Key."

echo "To use Elementary OS AI you need to type the command chat in the Terminal then add a question using "ask". Example: chat "how install flatpak hub repository in Elementor OS?"

# Initialize Shell Genie
shell-genie init

**Comment by madhums** [comment]
I think we, as open source and ethical tech enthusiasts, must be really careful before integrating an AI like Chat-GPT or any other AI bot, as not all AI bot developers/organisations (like Microsoft, OpenAI, Google, etc.) have respected and consented with the data.

I think it's a nice idea but needs to be thought about thoroughly from a human and ethical perspective.

**Comment by davidak** [comment]
elementary OS is a privacy-respecting open source operating system

ChatGPT is a proprietary privacy-violating online-service

this does not fit at all!

i'm following open source LLM development and alignment research (e.g. Heuristic Imperatives) actively and it might result in a usable tool, but we NEED DESIGN of a tool that actually has some use for people. can we have a system service that can use different machine learning models locally on GPU / CPU which can be used by applications? then GUI applications for elementary OS can use it

i don't see any use of a CLI chat program using the OpenAI API for elementary OS users. they can better use ChatGPT in the browser!
if YOU want to use something like that, just install it like on any other linux-based OS

**Reply by cosmic-zip** [reply]
A different option is gpt4all. I've used it for a while, and with the llama model 7b, I get results that are comparable to GPT 3.5, which is what I use the most for coding.

**Reply by christosalexiou** [reply]
running inference locally has implication's on the system's performance. vertically integrating this with the desktop is not a good idea.

**Comment by darltrash** [comment]
I think something like Siri would be great but not under our current technology which is fairly heavy and computationally expensive, and OpenAI itself is quite a terrible company when it comes to respecting privacy as mentioned here by everyone before me
